test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -829,28 +1249,103 @@ test.describe( 'Block bindings', () => {
 				await imageBlockImg.click();
 
 				// Image src is the custom field value.
-				const imageSrc = await imageBlockImg.getAttribute( 'src' );
-				expect( imageSrc ).toBe( customFieldSrc );
+				await expect( imageBlockImg ).toHaveAttribute(
+					'src',
+					imageCustomFieldSrc
+				);
 
 				// Alt textarea is disabled and with the custom field value.
 				await expect(
-					page.getByLabel( variables.labels.imageAlt )
+					page
+						.getByRole( 'tabpanel', { name: 'Block' } )
+						.getByLabel( 'Alternative text' )
 				).toBeDisabled();
 				const altValue = await page
-					.getByLabel( variables.labels.imageAlt )
+					.getByRole( 'tabpanel', { name: 'Block' } )
+					.getByLabel( 'Alternative text' )
 					.inputValue();
-				expect( altValue ).toBe( variables.customFields.textValue );
+				expect( altValue ).toBe( 'Value of the text_custom_field' );
 
 				// Title input is enabled and with the original value.
-				await page.getByRole( 'button', { name: 'Advanced' } ).click();
+				const advancedButton = page
+					.getByRole( 'tabpanel', { name: 'Block' } )
+					.getByRole( 'button', {
+						name: 'Advanced',
+					} );
+				const isAdvancedPanelOpen =
+					await advancedButton.getAttribute( 'aria-expanded' );
+				if ( isAdvancedPanelOpen === 'false' ) {
+					await advancedButton.click();
+				}
 				await expect(
-					page.getByLabel( variables.labels.imageTitle )
+					page
+						.getByRole( 'tabpanel', { name: 'Block' } )
+						.getByLabel( 'Title attribute' )
 				).toBeEnabled();
 				const titleValue = await page
-					.getByLabel( variables.labels.imageTitle )
+					.getByRole( 'tabpanel', { name: 'Block' } )
+					.getByLabel( 'Title attribute' )
 					.inputValue();
 				expect( titleValue ).toBe( 'default title value' );
+
+				// Check the frontend uses the values of the custom fields.
+				await BlockBindingsUtils.setId( 'image-multiple-bindings' );
+				const postId = await BlockBindingsUtils.updatePost();
+				await page.goto( `/?p=${ postId }` );
+				const imageDom = page.locator( '#image-multiple-bindings img' );
+				await expect( imageDom ).toBeVisible();
+				await expect( imageDom ).toHaveAttribute(
+					'src',
+					imageCustomFieldSrc
+				);
+				await expect( imageDom ).toHaveAttribute(
+					'alt',
+					'Value of the text_custom_field'
+				);
+				await expect( imageDom ).toHaveAttribute(
+					'title',
+					'default title value'
+				);
 			} );
 		} );
 	} );
 } );
+
+class BlockBindingsUtils {
+	constructor( { page } ) {
+		this.page = page;
+	}
+
+	// Helper to add an anchor/id to be able to locate the block in the frontend.
+	async setId( testId ) {
```

Comment:
Why don't we just set the `id` when we insert the block?

Reply:
How can we do that? Can we add an `id` that persists in the frontend?

Reply:
I believe this is possible by adding the `anchor` attribute?

```js
await editor.insertBlock( {
  name: 'core/paragraph',
  attributes: {
    anchor: 'paragraph-id',
    content: 'My paragraph
  },
} );
```

Reply:
I made this change as part of [this other pull request](https://github.com/WordPress/gutenberg/pull/58785).